drv/transceivers-server/src/net_api.rs

```diff
@@ -0,0 +1,414 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Hardware revision-independent server code for UDP interactions
+//!
+//! This is in a separate module to avoid polluting `main.rs` with a bunch of
+//! imports from the `transceiver_messages` crate; it simply adds more functions
+//! to our existing `ServerImpl`.
+use crate::ServerImpl;
+use drv_sidecar_front_io::transceivers::{
+    FpgaController, FpgaPortMasks, PortLocation,
+};
+use hubpack::SerializedSize;
+use task_net_api::*;
+use transceiver_messages::{
+    message::*,
+    mgmt::{MemoryRegion, UpperPage},
+    Error, ModuleId,
+};
+use userlib::hl::sleep_for;
+
+/// Convert from the over-the-network type to our local port mask type
+fn get_mask(m: ModuleId) -> Result<FpgaPortMasks, Error> {
+    let fpga_ports: u16 = m.ports.0;
```

Comment:
I'd rather keep the `transceiver-messages` types isolated to this one file.

(Otherwise, we'd need to add a dependency on `transceiver-messages` to the low-level `sidecar-front-io` crate, which is a bit of a leak across abstraction layers)